src/interchange/src/protobuf.rs

```diff
@@ -94,19 +94,26 @@ impl DecodedDescriptors {
 pub struct Decoder {
     message_descriptor: MessageDescriptor,
     packer: Row,
+    confluent_wire_format: bool,
 }
 
 impl Decoder {
     /// Constructs a decoder for a particular Protobuf message.
-    pub fn new(descriptors: DecodedDescriptors) -> Self {
+    pub fn new(descriptors: DecodedDescriptors, confluent_wire_format: bool) -> Self {
         Decoder {
             message_descriptor: descriptors.message_descriptor,
             packer: Row::default(),
+            confluent_wire_format,
         }
     }
 
     /// Decodes the encoded Protobuf message into a [`Row`].
-    pub fn decode(&mut self, bytes: &[u8]) -> Result<Option<Row>, anyhow::Error> {
+    pub fn decode(&mut self, mut bytes: &[u8]) -> Result<Option<Row>, anyhow::Error> {
+        if self.confluent_wire_format {
+            // We don't care about the schema id for now
+            let (_, adjusted_bytes) = crate::confluent::extract_schema_id(bytes, true)?;
```

Comment:
We should in fact verify that the schema ID/schema matches the one that we got in purify. See all the action that we have around `SchemaCache` in the related `Avro` code.

At this point we don't want to support schema evolution so most of that work is unnecessary, but initial validation is important: so basically check on first access if we have a schema that matches the purified one, and after that check if the Schema ID has changed and error out if it has.